aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/ApiMocking.java

```diff
@@ -0,0 +1,150 @@
+
+            // Pass back a ModelWithMetadata. Model is the one provided.
+            ModelMetadata metadata =
+                new ModelMetadata(model.getId(), false, 1, new Temporal.Timestamp());
+            ModelWithMetadata<M> modelWithMetadata = new ModelWithMetadata<>(model, metadata);
+            Consumer<GraphQLResponse<ModelWithMetadata<M>>> onResult =
+                invocation.getArgument(indexOfResultConsumer);
+            onResult.accept(new GraphQLResponse<>(modelWithMetadata, Collections.emptyList()));
+
+            // Technically, create() returns a Cancelable...
+            return null;
+        }).when(mockApi).mutate(
+            argThat(getMatcherFor(model)), // Match the id of the model passed in to the function.
+            any(), // onResponse
+            any() // onFailure
+        );
+    }
+
+    /**
+     * Mock the results of successful query operation triggered by the DataStore.
+     * @param mockApi Mock object of type {@link GraphQLBehavior}
+     * @param models A list of models to be returned in the result.
+     * @param <M> The model type.
+     */
+    @SuppressWarnings("unchecked")
+    public static <M extends Model> void mockSuccessfulSyncQuery(GraphQLBehavior mockApi, M... models) {
+        doAnswer(invocation -> {
+            final int onResultsIndex = 1;
+            // Transform from an array of Model types to, to a list of ModelWithMetadata<M extends Model>
+            List<ModelWithMetadata<M>> syncQueryResults =
+                Observable.fromArray(models)
+                          .map(model -> {
+                              ModelMetadata metadata = new ModelMetadata(model.getId(),
+                                                                         false,
+                                                                         1,
+                                                                         Temporal.Timestamp.now());
+                              ModelWithMetadata<M> modelWithMetadata = new ModelWithMetadata<>(model, metadata);
+                              return modelWithMetadata;
+                          })
+                          .toList()
+                          .blockingGet();
+
+            // The callback function to be invoked with the results
+            Consumer<GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>>> onResults =
+                invocation.getArgument(onResultsIndex);
+            onResults.accept(new GraphQLResponse<PaginatedResult<ModelWithMetadata<M>>>(
+                new PaginatedResult<ModelWithMetadata<M>>(syncQueryResults, null),
+                Collections.emptyList()
+            ));
+            return null;
+        }).when(mockApi).query(
+            any(), // Request
+            any(), // onResponse
+            any()  // onFailure
+        );
+    }
+
+    private static <M extends Model> ArgumentMatcher<GraphQLRequest<M>> getMatcherFor(M model) {
```

Comment:
Maybe just take the model ID as an arg, and rename it to:
```suggestion
    private static <M extends Model> ArgumentMatcher<GraphQLRequest<M>> requestContainsModelId(String modelId) {
```